src/Illuminate/Database/Query/Builder.php

```diff
@@ -1811,6 +1811,72 @@ protected function addDynamic($segment, $connector, $parameters, $index)
         $this->where(Str::snake($segment), '=', $parameters[$index], $bool);
     }
 
+    /**
+     * Slice where conditions at the given offset and add them to the query as a nested condition.
+     *
+     * @param  array  $whereSlice
+     * @return void
+     */
+    protected function groupWhereSlice($whereSlice)
```

Comment:
Copy from `Eloquent/Builder` with first parameter removed since we are now in the Query class itself.

Reply:
This method doesn't exists on `Eloquent/Builder` atm? Can you point out where it is?

Reply:
oops: `groupWhereSliceForScope` renamed to `groupWhereSlice` since it is not about scopes specifically anymore.